### PR TITLE
Improved README remote-signer, taquito and michelson-encoder

### PR DESCRIPTION
**Pull request by roxaneletourneau**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/README.md [hunk 1]
```diff
@@ -12,9 +11,140 @@ The `@taquito/taquito` package contains higher level functionality that builds u
...
+import { TezosToolkit } from '@taquito/taquito';
+
+const Tezos = new TezosToolkit('https://YOUR_PREFERRED_RPC_URL');
+```
+
+It is also possible to instantiate the `TezosToolkit` with a class that implements the `RpcClientInterface`. See the `RpcClientCache` from the `@taquito/rpc` package as an example that provides caching functionality.
+
+### Choosing between the contract or the wallet APIs
+
+In most cases, you want to use the Wallet API when you give the users of your dapp the freedom to choose the wallet of their choice to interact with it. The Contract API is more suited for back-end applications and forging/signing offline (for example, using the inMemorySigner). You would also use the Contract API to build a wallet.
```
Comment (michaelkernaghan) [inline packages/taquito/README.md:40]: to interact with it.   =>  to interact with.

File: packages/taquito/README.md [hunk 2]
```diff
@@ -12,9 +11,140 @@ The `@taquito/taquito` package contains higher level functionality that builds u
...
+
+**Configure a signer to use the Contract API**
+
+Sending operation using the Contract API requires a signer to be configured. Taquito provides different signers implementation (e.g. see the `taquito/remote-signer`, `taquito/signer` and `taquito/legder-signer`). Here is an example using the `InMemorySigner`:
```
Comment (michaelkernaghan) [inline packages/taquito/README.md:44]: operation  -> operations

Comment (michaelkernaghan) [inline packages/taquito/README.md:44]: different signers implementation -> different signer implementations

File: packages/taquito/README.md [hunk 3]
```diff
@@ -12,9 +11,140 @@ The `@taquito/taquito` package contains higher level functionality that builds u
...
+
+Tezos.setProvider({ signer: await InMemorySigner.fromSecretKey('edsk...') });
+
+// Using the contract API, the follwing operation is signed using the configured signer:
+await Tezos.contract.transfer({ to: publicKeyHash, amount: 2 });
+```
+
+**Configure a wallet to use the Wallet API**
+
+Sending operation using the Wallet API requires a wallet to be configured. The wallet API supports different kinds of wallets. For example, the `BeaconWallet` from the `@taquito/beacon-wallet` can be used. Use the `setWalletProvider` method of the `TezosToolkit` to set the wallet and refer to the `@taquito/beacon-wallet` for specific configuration:
```
Comment (michaelkernaghan) [inline packages/taquito/README.md:60]: operation -> operations

**Review by dsawali — APPROVED** [review]

**Pull request merged**